Libraries/LibEmail/IMAPClient.cpp

```diff
@@ -0,0 +1,284 @@
+#include <AK/StringBuilder.h>
+#include <LibEmail/IMAPClient.h>
+#include <ctype.h>
+
+namespace LibEmail {
+
+static ResponseStatus get_response_status(StringView response)
+{
+    if (response.empty())
+        return ResponseStatus::Bad;
+
+    const auto right_before_status = response.find_first_of(' ');
+    if (!right_before_status.has_value())
+        return ResponseStatus::Bad;
+
+    if (response.substring_view(right_before_status.value() + 1, 2) == "OK") {
+        return ResponseStatus::Ok;
+    } else if (response.substring_view(right_before_status.value() + 1, 2) == "NO") {
+        return ResponseStatus::No;
+    } else if (response.substring_view(right_before_status.value() + 1, 3) == "BAD") {
+        return ResponseStatus::Bad;
+    } else {
+        return ResponseStatus::FailedToSend;
+    }
+}
+
+static String get_message_field_string(Field value)
+{
+    switch (value) {
+    case Field::InternalDate:
+        return "INTERNALDATE";
+    case Field::Flags:
+        return "FLAGS";
+    case Field::UID:
+        return "UID";
+    case Field::Envelope:
+        return "ENVELOPE";
+    case Field::BodyText:
+        return "BODY[TEXT]";
+    }
+}
+
+IMAPClient::IMAPClient(StringView address, int port)
+    : m_socket(address, std::to_string(port))
+{
+    bool connected = m_socket->connect(address, port);
+    dbg() << "Step 1: connection: ";
+    if (connected)
+        dbg() << "Connected successfully";
+    else
+        dbg() << "Failed to connect";
+
+    // Setup thread for receiving/queuing server responses
+    m_receive_thread = LibThread::Thread::construct(
+        [this] {
+            while (true) {
+                const std::string response { m_socket->receive(1000) };
+                if (response.empty())
+                    // For some reason servers sometimes send empty messages
+                    continue;
+                m_queue_lock.lock();
+                m_message_queue.push_back(response);
+                m_queue_lock.unlock();
+            }
+            return 0;
+        });
```

Comment:
This looks weird:
```cpp
        });
    m_receive_thread.start();
}
```
You could use this style instead:
```cpp
    m_receive_thread = LibThread::Thread::construct([this] {
        ...
    });
    m_receive_thread.start();
}
    
```

Reply:
just do what clang-format is doing - and sometimes it is doing that.

Reply:
Both styles are accepted by `clang-format`.

Reply:
.